test/registration/registration_api_test.rb

```diff
@@ -0,0 +1,79 @@
+require 'test_helper'
+require 'registration/registration_api'
+
+class RegistrationRegisterApiTest < Test::Unit::TestCase
+  include Rack::Test::Methods
+
+  def app
+    Proxy::Registration::Api.new
+  end
+
+  def setup
+    @foreman_url = 'http://foreman.example.com'
+    Proxy::SETTINGS.stubs(:foreman_url).returns(@foreman_url)
+  end
+
+  def test_global_register_template
+    stub_request(:get, "#{@foreman_url}/register").to_return(body: 'template')
+
+    get "/"
+    assert last_response.ok?
+    assert_match('template', last_response.body)
+  end
+
+  def test_global_register_template_with_args
+    stub_request(:get, "#{@foreman_url}/register?param=test").to_return(body: 'template')
+
+    get '/', { param: 'test' }
+    assert last_response.ok?
+    assert_match('template', last_response.body)
+  end
+
+  def test_host_register_template
+    stub_request(:post, "#{@foreman_url}/register").to_return(body: 'template')
+
+    post '/'
+    assert last_response.ok?
+    assert_match('template', last_response.body)
+  end
+
+  def test_host_register_template_with_args
+    stub_request(:post, "#{@foreman_url}/register?host=test.example.com").to_return(body: 'template')
+
+    post '/', { host: 'test.example.com' }
+    assert last_response.ok?
+    assert_match('template', last_response.body)
+  end
+
+  def test_global_401
+    stub_request(:get, "#{@foreman_url}/register?error=401").to_return(body: 'error404')
```

Comment:
This looks odd. It suggests there's a HTTP GET parameter error, but there really isn't.
```suggestion
    stub_request(:get, "#{@foreman_url}/register").to_return(body: 'error401', status: 401)
```

(Same below with host_401)